tests/test_templates_layers.py

```diff
@@ -0,0 +1,684 @@
+# Copyright 2018 Xanadu Quantum Technologies Inc.
+
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+
+#     http://www.apache.org/licenses/LICENSE-2.0
+
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""
+Unit tests for the :mod:`pennylane.template.layers` module.
+"""
+# pylint: disable=protected-access,cell-var-from-loop
+import pytest
+
+import logging as log
+import pennylane as qml
+from pennylane import numpy as np
+from pennylane.qnode import QuantumFunctionError
+from pennylane.plugins import DefaultGaussian
+from pennylane.templates.layers import (Interferometer,
+                                        CVNeuralNetLayers,
+                                        StronglyEntanglingLayers, StronglyEntanglingLayer,
+                                        RandomLayers, RandomLayer)
+from pennylane import RX, RY, RZ, CZ, CNOT
+log.getLogger('defaults')
+
+
+class DummyDevice(DefaultGaussian):
+    """Dummy device to allow Kerr operations"""
+    _operation_map = DefaultGaussian._operation_map.copy()
+    _operation_map['Kerr'] = lambda *x, **y: np.identity(2)
+
+
+TOL = 0.0001
```

Comment:
I've added a `tol` fixture in `conftest.py` if you want. This way, you can simply add `tol` to the test function argument.

Reply:
Thanks! Some tol parameters are fine tuned, but using the fixture for the others now.